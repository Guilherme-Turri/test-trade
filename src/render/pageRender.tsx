import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};