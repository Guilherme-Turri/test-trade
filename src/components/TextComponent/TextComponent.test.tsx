import { componentRender } from "../../render/componentRender";
import { screen } from '@testing-library/react'
import { TextComponent } from ".";

describe('<TextComponent />', () => {
  it('Should render on screen properly', () => {
    const { container } = componentRender(<TextComponent>Test</TextComponent>);
    expect(screen.getByText('Test')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
})