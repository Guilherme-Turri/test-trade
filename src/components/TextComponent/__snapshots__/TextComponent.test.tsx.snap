// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> Should render on screen properly 1`] = `
.c0 {
  color: #273036;
}

<div>
  <p
    class="c0"
  >
    Test
  </p>
</div>
`;
